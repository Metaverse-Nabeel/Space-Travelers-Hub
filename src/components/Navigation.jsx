/* eslint-disable no-nested-ternary */
/* eslint-disable react/jsx-closing-bracket-location */
import Container from 'react-bootstrap/Container';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import { NavLink } from 'react-router-dom';
import logo from '../assets/logo.png';
import NavCSS from '../styles/Navigation.module.css';

const Navigation = () => (
  <>
    <Navbar bg="light" variant="light">
      <Container className={NavCSS.navbarContainer}>
        <Navbar.Brand className={NavCSS.navbarBrand}>
          <img
            alt="Logo"
            src={logo}
            width="70"
            height="70"
            className="d-flex align-top"
            />
          Space Travelers&apos; Hub
        </Navbar.Brand>
        <Nav className={NavCSS.navigationLinks}>
          <NavLink
            to="/"
            style={({ isActive, isPending }) => ({
              paddingRight: isActive ? '1rem' : '',
              textDecoration: isActive ? 'underline' : '',
              color: isPending ? '#6abada' : '#0e7ca7',
            })}>
            Rockets
          </NavLink>
          <NavLink
            to="/missions"
            style={({ isActive, isPending }) => ({
              paddingRight: isActive ? '1rem' : '',
              textDecoration: isActive ? 'underline' : '',
              color: isPending ? '#6abada' : '#0e7ca7',
            })}>
            Missions

          </NavLink>
          <NavLink
            to="/profile"
            style={({ isActive, isPending }) => ({
              paddingRight: isActive ? '1rem' : '',
              textDecoration: isActive ? 'underline' : '',
              color: isPending ? '#6abada' : '#0e7ca7',
            })}>
            My Profile
          </NavLink>
        </Nav>
      </Container>
    </Navbar>
  </>
);

export default Navigation;
