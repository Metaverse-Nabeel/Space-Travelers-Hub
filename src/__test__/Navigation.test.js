import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navigation from '../components/Navigation';

it('Navigation Component renders correctly', () => {
  const tree = render(
    <Router>
      <Navigation />
    </Router>,
  );
  expect(tree).toMatchSnapshot();
});
