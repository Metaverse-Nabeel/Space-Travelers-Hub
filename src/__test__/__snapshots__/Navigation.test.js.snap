// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand navbar-light bg-light"
      >
        <div
          class="navbarContainer container"
        >
          <span
            class="navbarBrand navbar-brand"
          >
            <img
              alt="Logo"
              class="d-flex align-top"
              height="70"
              src="logo.png"
              width="70"
            />
            Space Travelers' Hub
          </span>
          <div
            class="navigationLinks navbar-nav"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
              style="text-decoration: underline; color: rgb(14, 124, 167);"
            >
              Rockets
            </a>
            <a
              class=""
              href="/missions"
              style="color: rgb(14, 124, 167);"
            >
              Missions
            </a>
            <a
              class=""
              href="/profile"
              style="color: rgb(14, 124, 167);"
            >
              My Profile
            </a>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand navbar-light bg-light"
    >
      <div
        class="navbarContainer container"
      >
        <span
          class="navbarBrand navbar-brand"
        >
          <img
            alt="Logo"
            class="d-flex align-top"
            height="70"
            src="logo.png"
            width="70"
          />
          Space Travelers' Hub
        </span>
        <div
          class="navigationLinks navbar-nav"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
            style="text-decoration: underline; color: rgb(14, 124, 167);"
          >
            Rockets
          </a>
          <a
            class=""
            href="/missions"
            style="color: rgb(14, 124, 167);"
          >
            Missions
          </a>
          <a
            class=""
            href="/profile"
            style="color: rgb(14, 124, 167);"
          >
            My Profile
          </a>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
